api/types.proto

```diff
@@ -281,15 +281,40 @@ message UpdateConfig {
 	enum FailureAction {
 		PAUSE = 0;
 		CONTINUE = 1;
-		// TODO(aaronl): Add ROLLBACK as a supported failure mode.
-		// (#486)
+		ROLLBACK = 2;
 	}
 
 	// FailureAction is the action to take when an update failures.
-	// Currently, a failure is defined as a single updated task failing to
-	// reach the RUNNING state. In the future, there will be configuration
-	// to define what is treated as a failure (see #486 for a proposal).
 	FailureAction failure_action = 3;
+
+	// MonitoringPeriod indicates how long to monitor a task for failure
+	// after it is created. If the task fails by ending up in one of the
+	// states REJECTED, COMPLETED, or FAILED, within MonitoringPeriod
+	// from its creation, this counts as a failure. If it fails after
+	// MonitoringPeriod, it does not count as a failure. If
+	// MonitoringPeriod is unspecified, a default value will be used.
+	Duration monitoring_period = 4;
```

Comment:
Changed to `monitor`.

Feel like this might be a little terse, but I see the point about flag names.